### PR TITLE
fix for #152

### PR DESCRIPTION
**Pull request by innin**
added validation of like to posts controller

### CODE REVIEW

File: app/controllers/api/v1/PostsController.js [hunk 1]
```diff
@@ -90,6 +90,7 @@ exports.addController = function(app) {
       return res.status(401).jsonp({ err: 'Not found' })
 
     models.Post.getById(req.params.postId)
+      .then(function(post) { return post.validateLikeOrUnlike('like', req.user.id) })
```
Comment (epicmonkey) [inline app/controllers/api/v1/PostsController.js]: Validation should never be a part of controller, let's push it a little further and extract to User model. The errors you saw in your last pull request #159 are OK. You may just want to update unit/functional tests to cover this.

**Comment by innin** [comment]
Ok, but there is validation call in this controller in `PostsController.create ` method:
```javascript
      return FeedFactory.findByUsername(username)
        .then(function(feed) {
          return feed.validateCanPost(req.user)
        })
        .then(function(feed) {
          return feed.getPostsTimelineId()
        })
```
Validator `validateLikeOrUnlike` itself is defined in Posts model, not in controller, just like  `validateCanPost`  validator. So I thought this is a normal practice.

Anyway I think I will need to look more deeply into how the project works before doing something and making pull-requests, so if you will make this fix in proper way and commit it, this will be helpful for me as well. Thanks!

**Comment by epicmonkey** [comment]
That's true, `validateCanPost` and some other `validate` methods are being called in controllers, though eventually it's going to be an extra authorization layer (for the reference, discussion our with yole https://github.com/pepyatka/pepyatka-server/pull/130/files#r28198540), that does not belong neither to models, nor controllers. Story is not created yet.

File: app/models/post.js [hunk 2]
```diff
@@ -127,6 +127,17 @@ exports.addModel = function(database) {
...
           })
         })
+        // decrement likes counter for users who liked this post
+        .then(function() {
+          return that.getLikeIds()
+        })
+        .then(function(userIds) {
+          return Promise.map(userIds, function(userId) {
+            return models.Stats.findById(userId).then(function(stats) {
+              return stats.removeLike()
```
Comment (epicmonkey) [inline app/models/post.js]: Let's add a new method Stats#removeLikes(count).

File: test/functional/posts.js [hunk 3]
```diff
@@ -152,7 +152,16 @@ describe("PostsController", function() {
           res.body.should.be.empty
           $should.not.exist(err)
 
-          done()
+          request
+            .post(app.config.host + '/v1/posts/' + post.id + '/like')
+            .send({ authToken: authToken })
+            .end(function(err, res) {
+              err.should.not.be.empty
+              err.status.should.eql(403)
```
Comment (epicmonkey) [inline test/functional/posts.js]: Let's improve this a little and check exceptions text message, e.g.
```
err.response.error.should.have.property('text')
JSON.parse(err.response.error.text).err.should.eql('Passwords do not match')
```

File: app/models/post.js (hunk 2 above)
Comment (epicmonkey) [inline app/models/post.js]: Per discussion -- I was wrong, it's not relevant for this case

File: test/functional/posts.js (hunk 3 above)
Comment (innin) [inline test/functional/posts.js]: added these checks. And synced my changes with latest changes with pepyatka:development

**Comment by epicmonkey** [comment]
Fixes #152, #156, and https://github.com/pepyatka/pepyatka-html/issues/148

**Pull request merged**